src/libraries/System.Private.CoreLib/src/System/Threading/WaitHandle.Windows.cs

```diff
@@ -134,12 +150,22 @@ private static int SignalAndWaitCore(IntPtr handleToSignal, IntPtr handleToWaitO
                 startTime = Environment.TickCount64;
             }
 
+#if !MONO
```

Comment:
MONO can have an empty implementation of `CheckForPendingInterrupt` to avoid these ifdefs.